lib/internal/bootstrap_node.js

```diff
@@ -411,8 +411,38 @@
 
   // Load preload modules
   function preloadModules() {
-    if (process._preload_modules) {
-      NativeModule.require('module')._preloadModules(process._preload_modules);
+    var allPreloads = process._preload_modules;
+    var isSetUidRoot = false;
+
+    if (process.platform !== 'win32') {
+      isSetUidRoot = process.getuid() !== process.geteuid() ||
+        process.getgid() !== process.getegid();
+    }
+
+    if (!isSetUidRoot && process.env.NODE_PRELOAD) {
+      const path = NativeModule.require('path');
+
+      const addPreload = (nmDir, preloads) => {
+        const addNmDir = (m) => {
+          return path.isAbsolute(m) ? m : path.join(nmDir, m);
+        };
+
+        if (preloads && preloads.length > 0) {
+          preloads = preloads.map((m) => addNmDir(m.trim()))
+              .filter((m) => (m.length > nmDir.length && m.startsWith(nmDir)));
+
+          if (preloads.length > 0) {
+            allPreloads = (allPreloads || []).concat(preloads);
+          }
+        }
+      };
+
+      const globalNm = path.join(process.argv[0], '../../lib/node_modules/');
```

Comment:
At the moment I am waiting for an approach that everyone can at least not put a -1 on first.  If @nodejs/ctc have an approach that's acceptable, I'd implement that.